### PR TITLE
Get Started updates

### PR DESCRIPTION
**Pull request by omesser**
https://mlem-ai-simplify-get-st-gwz7i9.herokuapp.com/doc/get-started

I've identified some issues which in my mind make the get started not as appealing and simple to understand as it could be

- The get started workflows today are "actions" - applying/serving/building/deploying. New users might not understand what those mean, in what order should they "put the pieces to gether" and this might lose apeal.
Some questions I can imagine here for new ysers:
  - "where should I start? I obviously care about all of those, I think"
  - "deploying vs serving? sounds the same"
  - "what does applying a model mean"?
  - "does it make sense to learn about deploying without building? building before applying?"

- "productionization" term is not a clear / common enough in my mind. It's a big catch-all term. it's the umbrella for all current get-started guides.

My Suggestion here - let's have full scenarios drive the GS structure - so less scenarios (2), more stages in each

# Suggested Structure:
	- Model Prediction
		- loading model and predicting in code
		- batch scoring
	- Deploying and serving models
		- running a local server
		- building/packaging models
		- deploying models to cloud

### CODE REVIEW

**Comment by jorgeorpinel** [comment]
Great Qs. Relates to discussion in https://github.com/iterative/mlem.ai/issues/58#issuecomment-1305806996.

**Comment by jorgeorpinel** [comment]
> Suggested Structure:
- basic model management ... saving and loading ... model inferencing and batch scoring
- Deploying and serving models ... running a local server ... building/packaging ... deploying to cloud

For the record, what I originally recommended was similar:

```
i. Capturing models (saving + packaging?)
ii. Productionizing (applying, serving, deploying)
```

Though I also think that the saving part is pretty good for the index of the section, as [is now](https://mlem.ai/doc/get-started). (Does that answer "where should I start?" or do we mean"where to go after saving?")

File: content/docs/get-started/index.md [hunk 1]
```diff
@@ -4,13 +4,14 @@ description: 'Learn how you can use MLEM to easily manage and deploy models'
...
-(if not, you can follow the instructions below).
+To get started, we'll start with installing the `mlem` python library containing
+the sdk and cli. For the rest of the guide, we assume MLEM is already
+[installed](/doc/install) in your active Python environment, as well as some
+other useful packages - `pandas`, `sklearn`, `fastapi`, `uvicorn` and `docker`.
 
 <details>
 
-### ⚙️ Expand for setup instructions
+### ⚙️ Expand for set-up instructions
```
Comment (jorgeorpinel) [inline content/docs/get-started/index.md:14]: All these preparatory steps can be hidden (if really needed) in the `<details>` section below. In fact they're all there already (except Docker?) so this seems redundant.

File: content/docs/get-started/index.md [hunk 2]
```diff
@@ -23,145 +24,30 @@ $ source .venv/bin/activate
...
-The `.mlem` file is a bit long, but it contains all the metadata we need to use
-the model later:
+- **[Basic Model Management](/doc/get-started/management)** - Save & load models
+  in Python and get predictions in the command line.
 
-1. Model methods: `predict` and `predict_proba`
-2. Input data schema: describes the data frame (Iris dataset)
-3. Requirements: `sklearn` and `pandas`, with specific versions
+- **[Serving models](/doc/get-started/deploying-and-serving)** - Package your
+  model, create a model server for online serving, and deploy it on the cloud.
```
Comment (jorgeorpinel) [inline content/docs/get-started/index.md:40]: I do like these [Scenarios](https://mlem-ai-simplify-get-st-gwz7i9.herokuapp.com/doc/get-started#scenarios) (use cases? workflows?) more than the current, low-level list of "[Productionization](https://mlem.ai/doc/get-started#productionization)" methods. But I'd still have them after the Saving models section (which lets people start using MLEM right away and hopefully hooked).

File: content/docs/get-started/management.md [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+Let's see what we got when we saved the `rf` model with MLEM above.
+
+```cli
+$ tree models/
+models
+├── rf
+└── rf.mlem
+```
+
+The model binary was saved to `models/rf`, along with some metadata about it in
```
Comment (jorgeorpinel) [inline content/docs/get-started/management.md:133]: This also I think was better closer to the Saving models section (in the GS index). In fact I'd make it part of that section (no separate header) and basically keep just one sentence and the dir tree. Everything else seems like implementation `<details>` to hide or move to a guide (linked).

File: content/docs/get-started/management.md [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+### Why do it the MLEM way
+
+Saving models to files or loading them back into python objects may seem like a
+deceptively simple task at first. For example, `pickle` and `torch` libraries
+can serialize/deserialize model objects to/from files, but we will see that MLEM
+adds some "special sauce" in the form of metadata files that will help us a lot
+down the line in the heavier operations like packaging and serving of the models
+in various ways. MLEM allows us to automate a lot of the pain points we would
+hit later on in our ML workflow by codifying metadata about your models (and
+other objects) and intelligently using it later on.
```
Comment (jorgeorpinel) [inline content/docs/get-started/management.md:120]: BTW, great info but too verbose for the Get Started. We fight for every word in this section. Let's move it to a guide somewhere?

**Review by jorgeorpinel — COMMENTED** [review]
> the saving part is pretty good for the index of the section, as [is now](https://mlem.ai/doc/get-started).

Comparing with the new index page, my conviction that the existing one is preferable increases. Specific reasons:

**Comment by jorgeorpinel** [comment]
I love the intention in this PR however other than the changes I reviewed above, we're mainly shuffling and regrouping existing content. The grouping is probably good but should be combined with summarizing things because in this proposal one of the pages is pretty short and the last one kind of long (they don't seem to be at the same conceptual level). We also add notes and paragraphs making it even longer which worries me.

Get Starteds are tricky 😓

**Comment by aguschin** [comment]
Thanks, @omesser. The suggestion is really good. I can see why it can be easier for a new user to consume.

I need to think some time about this restructuring but would like to discuss another option with you meanwhile.

Considering @jorgeorpinel feedback about moving `tree models/` details into index and maybe moving `building` into another section, I can suggest another options that builds on top what you did:

```
- index
	- saving models
	- "tree models/" part, about mlem metafiles + your motivation about "mlem way"
- loading models in Python runtime / basic model management
	- loading models
- Batch scoring
	- model inferencing and batch scoring
	- building/packaging models for batch scoring
- Deploying and serving models
	- running a local server
	- (short reference only) building/packaging models
	- deploying models to cloud
```

What I like about this suggestion:
1. we split GS by "use cases": Python runtime, Batch scoring, REST API deployment. This should be a clear fit for users.
2. model saving is in the "index" - you can go to any "use case" after index, if in rush.

What I don't like:
1. "loading models in Python runtime" is very short. Maybe easier to join with "index" then.
2. In this PR, user most probably will go through all of sections. Not sure it will happen with my suggestion.

WDYT? @omesser @jorgeorpinel

File: content/docs/get-started/management.md (hunk 4 above)
Comment (aguschin) [inline content/docs/get-started/management.md:120]: > We also add notes and paragraphs making it even longer which worries me

@jorgeorpinel, does your comment refers to the sections like these? I understand your intention, but I think those make things much clear. Besides, we "sell" MLEM. We can hide them in <details> as an option, but I believe these can be very helpful. WDYT?

File: content/docs/get-started/index.md (hunk 2 above)
Comment (omesser) [reply content/docs/get-started/index.md:40]: @jorgeorpinel My initial response here was going to be "but there is symmetry between saving and loading that feels awkward to break"
But after discussing with @shcheklein I want to maybe try and reduce this to 1 workflow/scenarios and significantly shorten it, this is why I'm dilly-dallying with getting back to this for another pass 😆

Comment (omesser) [reply content/docs/get-started/index.md:40]: I'm going to push a pass of fixes here and that would be one possible suggestion, and open another PR with an alternative approach (much more condensed)

File: content/docs/get-started/management.md (hunk 4 above)
Comment (jorgeorpinel) [reply content/docs/get-started/management.md]: > > We also add notes and paragraphs making it even longer
>
> does your comment refers to the sections like these?

Yes. The intention is good but we need the GS to be as brief and direct as possible. If we need to clarify something then it probably wasn't written properly in the first place (or there's an underlying product design issue).

p.s. sorry for late reply. Will check #238 post-merge.

**Pull request closed without merging**